### PR TITLE
Add CEL probe

### PR DESCRIPTION
**Pull request by thetechnick**
### Summary
Add a Common Expression Language probe to allow even the most complex probes to be defined.
### Change Type
New Feature

### Check List Before Merging

- [x] This PR passes all pre-commit hook validations.
- [x] This PR is fully tested and regression tests are included.
- [ ] Relevant documentation has been updated.

### CODE REVIEW

File: internal/controllers/objectdeployments/new_revision_reconciler.go [hunk 1]
```diff
@@ -116,3 +121,10 @@ func (r *newRevisionReconciler) newObjectSetFromDeployment(
 	}
 	return newObjectSet, nil
 }
+
+func latestRevisionNumber(prevObjectSets []genericObjectSet) int64 {
```
Comment (geoberle) [inline internal/controllers/objectdeployments/new_revision_reconciler.go:125]: is this related to the CEL prober?

File: internal/probing/cel_test.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+					"metadata": map[string]interface{}{
+						"name": "nothans",
+					},
+				},
+			},
+			success: false,
+		},
+		{
+			name:    "OpenShift Route success simple",
+			rule:    `self.status.ingress.all(i, i.conditions.all(c, c.type == "Ready" && c.status == "True"))`,
```
Comment (geoberle) [inline internal/probing/cel_test.go:55]: ❤️

**Review by geoberle — APPROVED** [review]
LGTM from my limited point of view.
thank you for this feature. this helps a lot!

File: internal/controllers/objectdeployments/new_revision_reconciler.go (hunk 1 above)
Comment (thetechnick) [reply internal/controllers/objectdeployments/new_revision_reconciler.go:125]: Nope, 2nd commit is a bugfix that fixes a problem due to template schema changing.
